### DISCUSSION TITLE
Installation fails with "ERROR updating files"

### DISCUSSION THREAD

**Discussion by Denperidge** [datenstrom/community#653] [Report a bug] [answered]
Hello there!
I've seem to be able to break a fresh Yellow setup because I first extracted the files on Windows, and then copied them to a Linux server. Directly wget'ting from the server fixed the issue.
The error it caused was the following (this was shown when running php yellow.php): `ERROR updating files: Can't open file 'system/extensions/install-languages.zip'!`.
This is thus presumably an issue with permissions. Removing the extensions "fixed" the problem (as in, the error was no longer displayed) but this causes setup to be skipped, which isn't much better.

This isn't necessarily a gigantic problem - as this is easily circumvented - but it might be worth investigating if this happens in other peoples situations too, and - if so - adding an entry to the troubleshooting page.

**Comment by schulle4u** [comment]
Insufficient write permissions or wrong ownership is the first problem we cover in our [troubleshooting guide](https://datenstrom.se/yellow/help/troubleshooting). Can you provide more details about your setup and how you transfered the files to your server? Otherwise everything is guesswork.

**Reply by Denperidge** [reply]
My apologies, was indeed a bit insufficient! I guess the thing that threw me off mostly was that in the web-ui, it said that language 'en' was not installed (which I also forgot to write down in the original post, my apologies). Running Windows 10 locally, used WinSCP/FTP to transfer to Debian Debian 8.3.0-6 installation.

Weirdly enough I haven't been able to recreate it anymore, so maybe it was a haphazard/bad-luck kinda deal? Merely wanted to post in case it would happen to other people as well, that there'd be more cases/info to go by.

**Reply by markseuffert** [reply]
Thanks for your feedback, could be a corrupted file that happened during file transfer.

**Comment by markseuffert** [comment, marked as answer]
I could reproduce the problem and updated the troubleshooting page. When you upload files with FTP it may happen that the data transfer is interrupted, then installation files are missing or with zero bytes on the web server. The new code will detect this and show the following error message: `Datenstrom Yellow requires complete upload`.  

Should be fixed now, please try it out.
